src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/JunctionTreeLinkedDeBruinGraph.java

```diff
@@ -452,25 +467,43 @@ private void threadSequenceForJuncitonTree(final SequenceForKmers seqForKmers) {
 
         // loop over all of the bases in sequence, extending the graph by one base at each point, as appropriate
         MultiDeBruijnVertex lastVertex = startingVertex;
-        int kmersPastSinceLast = 0;
+        boolean hasToRediscoverKmer = false;
+        int kmersPastSinceLastNullVertex = 0;
         for ( int i = startPos + 1; i <= seqForKmers.stop - kmerSize; i++ ) {
-            final MultiDeBruijnVertex vertex;
-            if (kmersPastSinceLast == 0) {
+            MultiDeBruijnVertex vertex;
+            if (!hasToRediscoverKmer) {
                 vertex = extendJunctionThreadingByOne(lastVertex, seqForKmers.sequence, i, nodeHelper);
             } else {
                 Kmer kmer = new Kmer(seqForKmers.sequence, i, kmerSize);
                 vertex = kmerToVertexMap.get(kmer);
-                // TODO this might cause problems
-                if (vertex != null) {
-                   attemptToResolveThreadingBetweenVertexes(lastVertex, nodeHelper, vertex);
+            }
+            // we found a null path in the grpah
+            if (vertex == null) {
+                // if we are not in an error kmer try to extend the path anyway assuming its a one base mismatch
+                if (kmersPastSinceLastNullVertex == 0) {
+                    Set<MultiSampleEdge> outgoingEdges = outgoingEdgesOf(lastVertex);
+                    if (outgoingEdges.size()==1) {
+                        vertex = getEdgeTarget(outgoingEdges.stream().findFirst().get());
+                        kmersPastSinceLastNullVertex = 1;
+                    }
+                // if we dropped multiple bases from the read following an already missing path from the reference then we throw it away and try to catch the trail from kmers again
+                } else {
+                    hasToRediscoverKmer = true;
+                    nodeHelper.clear();
                 }
             }
+
             // If for whatever reason vertex = null, then we have fallen off the corrected graph so we don't update anything
             if (vertex != null) {
                 lastVertex = vertex;
-                kmersPastSinceLast = 0;
-            } else {
-                kmersPastSinceLast++;
+
+                // If we aren't in an error state do nothing, otherwise we increment the count of bases, if we are over kmersize then revert to a safety state
+                if (kmersPastSinceLastNullVertex > 0) {
+                    kmersPastSinceLastNullVertex++;
+                    if (kmersPastSinceLastNullVertex > kmerSize) {
+                        kmersPastSinceLastNullVertex = 0;
```

Comment:
Yes. I will try to clean up the comments to this effect: 
We want to "skip" over single base incongruities if we can since they mean some path was pruned from the graph. In order to do that we have a "unsafe state" mode and the normal mode. If we are missing our next edge then we attempt to skip one edge past in both and flip into the unsafe state. The idea is that if there was a pruned SNP error we want to try to walk over the next kmer size bases and if we find k. If we find k exact matching bases between the read and the graph then we have walked over an entire kmer. There is a potential issue with this code that we discussed. I will refactor this to check ahead of time for validity without touching the trees at all.